t/t2221-job-manager-limit-duration.t

```diff
@@ -0,0 +1,124 @@
+	cat >config/policy.toml <<-EOT &&
+	[policy.limits]
+	duration = "1m"
+	EOT
+	flux config reload
+'
+test_expect_success 'the limit-duration plugin can now be laoded' '
+	flux jobtap load .limit-duration
+'
+test_expect_success 'a job that exceeds policy.limits.duration is rejected' '
+	test_must_fail flux mini submit -t 1h /bin/true 2>duration.err &&
+	grep "exceeds policy limit of 1m" duration.err
+'
+test_expect_success 'a job that sets no explicit duration is accepted' '
+	flux mini submit /bin/true
+'
+test_expect_success 'a job that is under policy.limits.duration is accepted' '
+	flux mini submit -t 30s /bin/true
+'
+test_expect_success 'configure policy.limits.duration and queue durations' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.limits]
+	duration = "1h"
+	[queues.pdebug.policy.limits]
+	duration = "1m"
+	[queues.pbatch.policy.limits]
+	duration = "8h"
+	EOT
+	flux config reload
+'
+test_expect_success 'a no-queue job that exceeds policy.limits.duration is rejected' '
+	test_must_fail flux mini submit -t 2h /bin/true
+'
+test_expect_success 'but is accepted by a queue with higher limit' '
+	flux mini submit \
+	    --setattr=system.queue=pbatch \
+	    -t 2h \
+	    /bin/true
+'
+test_expect_success 'and is rejected when it exceeds the queue limit' '
+	test_must_fail flux mini submit \
+	    --setattr=system.queue=pbatch \
+	    -t 16h \
+	    /bin/true
+'
+test_expect_success 'a job that is under policy.limits.duration is accepted' '
+	flux mini submit -t 1h /bin/true
+'
+test_expect_success 'but is rejected on a queue with lower limit' '
+	test_must_fail flux mini submit \
+	    --setattr=system.queue=pdebug \
+	    -t 1h \
+	    /bin/true
+'
+test_expect_success 'configure policy.limits.duration and an unlimited queue' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.limits]
+	duration = "1h"
+	[queues.pdebug.policy.limits]
+	duration = 0
+	EOT
+	flux config reload
+'
+test_expect_success 'a job that is over policy.limits.duration is rejected' '
+	test_must_fail flux mini submit -t 2h /bin/true
+'
+test_expect_success 'but is accepted by the unlimited queue' '
+	flux mini submit \
+	    --setattr=system.queue=pdebug \
+	    -t 2h /bin/true
+'
+test_expect_success 'a job that sets no explicit duration is accepted by the unlimited queue' '
+	flux mini submit \
+	    --setattr=system.queue=pdebug \
+	    /bin/true
+'
+test_expect_success 'configure an invalid string duration limit' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.limits]
+	duration = "xyz123"
+	EOT
+	test_must_fail flux config reload
+'
+test_expect_success 'configure a duration limit of an invalid type' '
+	cat >config/policy.toml <<-EOT &&
+	[policy.limits]
+	duration = [ 0 ]
+	EOT
+	test_must_fail flux config reload
+'
+test_expect_success 'configure an invalid queue duration limit' '
+	cat >config/policy.toml <<-EOT &&
+	[queues.pdebug.policy.limits]
+	duration = "xyz123"
+	EOT
+	test_must_fail flux config reload
+'
```

Comment:
worthwhile to add test that also adds in `policy.jobspec.defaults.system.duration`?

Reply:
Well that uncovered a bug where I was using `json_real_value()` to parse a type = json_integer value and getting 0.0, so yes it was worthwhile!